monai/transforms/transforms.py

```diff
@@ -206,10 +207,112 @@ def __call__(self, img):
         return rotator(img)
 
 
-# if __name__ == "__main__":
-#     img = np.array((1, 2, 3, 4)).reshape((1, 2, 2))
-#     rotator = RandRotate90(prob=0.0, max_k=3, axes=(1, 2))
-#     # rotator.set_random_state(1234)
-#     img_result = rotator(img)
-#     print(type(img))
-#     print(img_result)
+class AffineGrid:
+    """
+    Affine transforms on the coordinates.
+    """
+
+    def __init__(self, rotate_params=None, shear_params=None, translate_params=None, scale_params=None):
+        self.rotate_params = rotate_params
+        self.shear_params = shear_params
+        self.translate_params = translate_params
+        self.scale_params = scale_params
+
+    def __call__(self, spatial_size, grid=None):
+        """
+        Args:
+            spatial_size (list or tuple of int): output grid size.
+            grid (ndarray): grid to be transformed. Shape must be (3, H, W) for 2D or (4, H, W, D) for 3D.
+        """
+        if grid is None:
+            grid = create_grid(spatial_size)
+
+        spatial_dims = len(spatial_size)
+        affine = np.eye(spatial_dims + 1)
+        if self.rotate_params:
+            affine = affine @ create_rotate(spatial_dims, self.rotate_params)
+        if self.shear_params:
+            affine = affine @ create_shear(spatial_dims, self.shear_params)
+        if self.translate_params:
```

Comment:
it is easier to fix the order of operations rather than provide a way of determining the order, but if we do fix the order I would say shift should be last. I think it would be expected that a transform would be applied to the image as it originally is located, if a shift precedes another operation this will distort the image relative to its new position.

Reply:
Maybe you can support user's definition of different params in order, and construct a "Sub transform chain" based on the params order.
Thanks.